src/jit/compiler.cpp

```diff
@@ -2506,20 +2506,14 @@ void Compiler::compSetProcessor()
 #elif defined(_TARGET_X86_)
     opts.compUseFCOMI = jitFlags.IsSet(JitFlags::JIT_FLAG_USE_FCOMI);
     opts.compUseCMOV  = jitFlags.IsSet(JitFlags::JIT_FLAG_USE_CMOV);
-    opts.compCanUseSSE2 = jitFlags.IsSet(JitFlags::JIT_FLAG_USE_SSE2);
 
-#if !defined(LEGACY_BACKEND) && !defined(FEATURE_CORECLR)
+#ifdef LEGACY_BACKEND
+    opts.compCanUseSSE2 = jitFlags.IsSet(JitFlags::JIT_FLAG_USE_SSE2);
+#else
     // RyuJIT/x86 requires SSE2 to be available: there is no support for generating floating-point
```

Comment:
BTW: The current desktop requires SSE2 - the CRT that it is compiled against assumes SSE2.